### ISSUE TITLE
Missing some scripts?

### ISSUE THREAD

**Issue by hexylena** [dnieuw/N-A-W#1]
Hey @dnieuw, I'm working at avans with Eefje Schrauwen on the NWA project, and she asked me to take a look at this. It looks pretty easy to get started with! But I think I'll be missing:

- `/mnt/appl/Porechop_customized/porechop-runner.py` are there customisations we need? Is it just the normal porechop?/
- `mnt/appl/bam2consensus.py`
- `/mnt/appl/extract_genes.R`

Would it be possible to include the relevant portions of these?

And lastly it might be interesting to have conda environments for these tools? https://www.nextflow.io/docs/latest/conda.html That would make this way easier for us to run.

**Comment by dnieuw** [comment]
Hi @hexylena,

Porechop_customized is a version of porechop which uses a specified list of barcodes for demultiplexing, which greatly improves its speed when demultiplexing many small fastq files, because it removes the step of automatic barcode detection. For the real-time data generation and demultiplexing which we do it is necessary to use this version. Also porechop doesn't parallelize very well, so splitting the fastq and running multiple parallel porechop instances is much faster than running a single porechop instance on 1 big file.

I'm not sure what you mean by including the relevant "portions".

We are working with conda indeed, and we have a conda environment in which we run this workflow. The environment simply contains the tools as described in the snakefile, and snakemake itself.

**Comment by hexylena** [comment]
Hey @dnieuw, thanks for the response!

Ok, is porechop_customized available from a public source? Or are these modifications you have locally, if so, would it be possible to share these with the Avans students so they can reproduce your existing pipeline?

Portions - Here I meant that it isn't clear for me where to find the following two scripts that would be needed for the pipeline:

- /mnt/appl/extract_genes.R
- /mnt/appl/bam2consensus.py

Conda - fantastic! Do you have an environment.yml file you could maybe include with this repository? I was thinking something more like the per-step isolated environments like here: https://snakemake.readthedocs.io/en/stable/snakefiles/deployment.html#integrated-package-management but a global environment makes sense too.

**Comment by dnieuw** [comment]
Dear @hexylena,

Sorry for the slow replies.

Porechop_customized is available from my github https://github.com/dnieuw/Porechop, I pushed a commit to reflect changes that I did not push yet to the repo.
I added the extract_genes.R and the bam2consensus.py to this repo.
I made a environment.yaml and added it to this github repo, it does contain some unnecessary tools which I added for convenience sake. I'm not so tidy when it comes to creating environments ;).

Kind regards,

David
